src/pages/guide/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,60 @@
+import { ReactNode, useRef, useState } from 'react';
+import clsx from 'clsx';
+import debounce from 'lodash/debounce';
+import { useDevice } from '@deriv-com/ui';
+import './Carousel.scss';
+
+type TCarouselProps = {
+    isControlVisible?: boolean;
+    items: ReactNode[];
+};
+
+const Carousel = ({ isControlVisible = true, items }: TCarouselProps) => {
+    const { isMobile } = useDevice();
+    const [currentIndex, setCurrentIndex] = useState(0);
+    const scrollRef = useRef<HTMLDivElement>(null);
+
+    const onScroll = debounce(() => {
+        const clientWidth = scrollRef.current?.clientWidth ?? 0;
+        const scrollLeft = scrollRef.current?.scrollLeft ?? 0;
+        const totalWidth = clientWidth + 24;
+        const selectedIndex = scrollLeft / totalWidth;
+
+        setCurrentIndex(selectedIndex);
+    }, 100);
+
+    return (
+        <div className='carousel'>
+            <div className='carousel__container' onScroll={onScroll} ref={scrollRef}>
+                {items.map((item, index) => {
+                    return (
+                        <div className='carousel__item' key={index}>
```

Comment:
`nit:` we shouldn't use index as the key, maybe we can use something from items instead